twoliter/src/cmd/build.rs

```diff
@@ -25,10 +33,94 @@ pub(crate) struct BuildVariant {
+                .context("The project was missing an sdk")?
+                .uri(),
+        )
+        .await?;
+        sdk_container
+            .cp(Path::new("twoliter/alpha/build/rpms"), packages_dir.path())
+            .await?;
+
+        let rpms_dir = project.project_dir().join("build").join("rpms");
+        fs::create_dir_all(&rpms_dir)?;
+        debug!("Copying rpms to build dir");
+        for maybe_file in fs::read_dir(packages_dir.path().join("rpms"))? {
+            let file = maybe_file?;
+            debug!("Copying '{}'", file.path().display());
+            fs::copy(file.path(), rpms_dir.join(file.file_name()))?;
+        }
+
+        let mut created_files = Vec::new();
+
+        let in_tree_sbkeys_dir = project.project_dir().join("sbkeys");
+        let sbkeys_dir: Box<dyn AsRef<Path>> = if in_tree_sbkeys_dir.is_dir() {
+            Box::new(in_tree_sbkeys_dir)
+        } else {
+            // Create a temp sbkeys directory
+            debug!("sbkeys dir not found. Creating a temporary directory");
+            let sbkeys_dir = TempDir::new().context("Unable to create a tempdir for sbkeys")?;
+            trace!("Using sbkeys tempdir: '{}'", sbkeys_dir.path().display());
+            sdk_container
+                .cp(Path::new("twoliter/alpha/sbkeys"), sbkeys_dir.path())
+                .await?;
+            Box::new(sbkeys_dir)
+        };
+
+        // TODO: Remove once models is no longer conditionally compiled.
+        // Create the models directory for the sdk to mount
+        let models_dir = project.project_dir().join("sources/models");
+        if !models_dir.is_dir() {
+            debug!("models source dir not found. Creating a temporary directory");
+            fs::create_dir_all(&models_dir.join("src/variant"))
+                .context("Unable to create models source directory")?;
+            created_files.push(models_dir)
+        }
+
+        // Hold the result of the cargo make call so we can clean up the project directory first.
+        let res = CargoMake::new(&project, &self.arch)?
+            .env("TWOLITER_TOOLS_DIR", tempdir.path().display().to_string())
+            .env("BUILDSYS_ARCH", &self.arch)
+            .env("BUILDSYS_VARIANT", &self.variant)
+            // TODO Support dynamically loading go modules based on packages
+            .env("GO_MODULES", "")
+            .env(
+                "BUILDSYS_SBKEYS_DIR",
+                (*sbkeys_dir).as_ref().join("sbkeys").display().to_string(),
+            )
+            .makefile(makefile_path)
+            .project_dir(project.project_dir())
+            .exec("build")
+            .await;
+
+        // Clean up all of the files we created
+        for file_name in created_files {
+            let added = Path::new(&file_name);
+            if added.is_file() {
+                remove_file(added).await?;
+            } else if added.is_dir() {
+                remove_dir_all(added).await?;
+            }
+        }
```

Comment:
I'm a bit skeptical of this. The frequency with which it will never run (a previous error occurred, or the user cancelled the program), and that all it contains is the directory `sources/models` makes me think it's fine to just leave these cryptic directories in place for alpha.

Reply:
Does it hurt to try to clean it up?

Reply:
I guess not, other than delaying the surprise of finding them until an error or cancelled execution has occurred.